cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1648,6 +1661,60 @@ String zkConnectionString(int connectToReplicas)  {
                     });
         }
 
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            KafkaBrokerConfigurationHelper kbch = new KafkaBrokerConfigurationHelper(kafkaSetOperations.getAdminClientProvider(), secretOperations);
+            return kafkaSetOperations.getAsync(namespace, KafkaCluster.kafkaClusterName(name))
+                .compose(sts -> {
+                    if (sts == null) {
+                        return Future.succeededFuture();
+                    }
+                    int replicas = kafkaCluster.getReplicas();
+                    List<Future> configFutures = new ArrayList<>(replicas);
+
+                    for (int podId = 0; podId < replicas; podId++) {
+                        int finalPodId = podId;
+                        Future<Admin> acFut = kbch.adminClient(sts, podId);
+                        configFutures.add(
+                                acFut.compose(ac -> {
+                                    if (ac == null) {
+                                        kafkaPodsUpdatedDynamically.put(finalPodId, null);
+                                        log.warn("Could not create admin client.");
+                                        return Future.succeededFuture();
+                                    } else {
+                                        Future<Map<ConfigResource, Config>> futCurrent = kbch.getCurrentConfig(finalPodId, ac);
+                                        Future<ConfigMap> futDesired = configMapOperations.getAsync(namespace, KafkaCluster.metricAndLogConfigsName(name));
+                                        log.debug("Determining kafka pod {} dynamic update ability", finalPodId);
+                                        return CompositeFuture.join(futCurrent, futDesired)
+                                                .compose(res -> {
+                                                    KafkaBrokerConfigurationDiff configurationDiff = new KafkaBrokerConfigurationDiff(res.result().resultAt(0), res.result().resultAt(1), kafkaCluster.getKafkaVersion(), finalPodId);
+
+                                                    if (configurationDiff.getDiff().asOrderedProperties().asMap().size() > 0) {
+                                                        log.debug("kafka changed, dynamically? : {}", !configurationDiff.cannotBeUpdatedDynamically());
```

Comment:
Just use the `reconciliation` as the prefix everywhere ;-). This log message looks weird again like some debugging left over.